src/components/HabitCard.tsx

```diff
@@ -1,20 +1,50 @@
 import React from "react";
-import Checkbox from "./Checkbox.js";
-import { connect } from "react-redux";
-import { updateStreak } from "../actions";
+import Checkbox from "./Checkbox";
+import { connect, ConnectedProps } from "react-redux";
+import { updateStreak } from "../store/actions";
 import { Link } from "react-router-dom";
 import {
   NO_STREAK,
   DISABLED_STREAK,
   ACTIVE_STREAK,
-  streakStatus
+  streakStatus,
 } from "../utils/streak";
-import { deleteHabit } from "../actions";
+import { deleteHabit } from "../store/actions";
 import "../css/main.css";
-import moment from "moment";
+import moment, { Moment } from "moment";
+import { State } from "../store/types";
+import { Habit } from "../models/habit";
+import { Streak } from "../models/streak";
 
-class HabitCard extends React.Component {
-  renderStreak(streak, endDate) {
+const mapState = (state: State, ownProps: OwnProps) => {
+  let id = ownProps.id;
```

Comment:
This variable is never mutated, so it should probably be `const`.

```suggestion
  const id = ownProps.id;
```